src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -430,23 +438,71 @@ export default abstract class ComparisonStore {
         },
     });
 
-    public readonly mutationEnrichmentProfiles = remoteData({
+    public readonly alterationEnrichmentProfiles = remoteData({
         await: () => [this.molecularProfilesInActiveStudies],
+        invoke: () => {
+            return Promise.resolve({
+                mutationProfiles: pickMutationEnrichmentProfiles(
+                    this.molecularProfilesInActiveStudies.result!
+                ),
+                structuralVariantProfiles: pickStructuralVariantEnrichmentProfiles(
+                    this.molecularProfilesInActiveStudies.result!
+                ),
+                copyNumberEnrichmentProfiles: pickCopyNumberEnrichmentProfiles(
+                    this.molecularProfilesInActiveStudies.result!
+                ),
+            });
+        },
+        onResult: profiles => {
+            if (profiles) {
+                if (!_.isEmpty(profiles.mutationProfiles)) {
+                    mutationGroup.forEach(eventType => {
+                        this.selectedMutationEnrichmentEventTypes[
+                            eventType
+                        ] = true;
+                    });
+                }
+                if (!_.isEmpty(profiles.structuralVariantProfiles)) {
+                    fusionGroup.forEach(eventType => {
+                        this.selectedMutationEnrichmentEventTypes[
+                            eventType
+                        ] = true;
+                    });
+                }
+                if (!_.isEmpty(profiles.copyNumberEnrichmentProfiles)) {
+                    cnaGroup.forEach(eventType => {
+                        this.selectedCopyNumberEnrichmentEventTypes[
+                            eventType
+                        ] = true;
+                    });
+                }
+            }
+        },
+    });
+
+    public readonly mutationEnrichmentProfiles = remoteData({
+        await: () => [this.alterationEnrichmentProfiles],
         invoke: () =>
             Promise.resolve(
-                pickMutationEnrichmentProfiles(
-                    this.molecularProfilesInActiveStudies.result!
-                )
+                this.alterationEnrichmentProfiles.result!.mutationProfiles
+            ),
+    });
+    //
+    public readonly structuralVariantProfiles = remoteData({
```

Comment:
i could have change all referencing code to access the single promise 
```
alterationEnrichmentProfiles.whatever
```
but decided it was better to tread lightly